test/com/google/enterprise/adaptor/DocIdSenderTest.java

```diff
@@ -347,74 +353,282 @@ public void getModifiedDocIds(DocIdPusher pusher) throws IOException, Interrupte
     }
   }
 
-  @Test
-  public void testPushGroupsNormal() throws Exception {
-    // Order of iteration matters
-    Map<GroupPrincipal, Collection<Principal>> groups
-        = new TreeMap<GroupPrincipal, Collection<Principal>>();
-    groups.put(new GroupPrincipal("g1"),
-        Arrays.asList(new UserPrincipal("u1"), new GroupPrincipal("g2")));
-    groups.put(new GroupPrincipal("g2"),
-        Arrays.asList(new UserPrincipal("u2"), new GroupPrincipal("g3")));
-    groups.put(new GroupPrincipal("g3"),
-        Arrays.asList(new UserPrincipal("u3"), new GroupPrincipal("g4")));
-    groups = Collections.unmodifiableMap(groups);
-
-    // I'm sorry.
-    List<List<Map.Entry<GroupPrincipal, Collection<Principal>>>> goldenGroups
-        = new ArrayList<List<Map.Entry<GroupPrincipal,
-          Collection<Principal>>>>();
-    {
-      List<Map.Entry<GroupPrincipal, Collection<Principal>>> tmp
-          = new ArrayList<Map.Entry<GroupPrincipal, Collection<Principal>>>();
-      tmp.add(new SimpleImmutableEntry<GroupPrincipal, Collection<Principal>>(
-          new GroupPrincipal("g1"), groups.get(new GroupPrincipal("g1"))));
-      tmp.add(new SimpleImmutableEntry<GroupPrincipal, Collection<Principal>>(
-          new GroupPrincipal("g2"), groups.get(new GroupPrincipal("g2"))));
-      goldenGroups.add(Collections.unmodifiableList(tmp));
-      goldenGroups.add(Collections.
-          <Map.Entry<GroupPrincipal, Collection<Principal>>>singletonList(
-          new SimpleImmutableEntry<GroupPrincipal, Collection<Principal>>(
-            new GroupPrincipal("g3"), groups.get(new GroupPrincipal("g3")))));
-      goldenGroups = Collections.unmodifiableList(goldenGroups);
+  private static Map<GroupPrincipal, Collection<Principal>> sampleGroups() {
+    return ImmutableMap.<GroupPrincipal, Collection<Principal>>of(
+        new GroupPrincipal("g1"), ImmutableList.<Principal>of(
+            new UserPrincipal("u1"), new GroupPrincipal("g2")),
+        new GroupPrincipal("g2"), ImmutableList.<Principal>of(
+            new UserPrincipal("u2"), new GroupPrincipal("g3")),
+        new GroupPrincipal("g3"), ImmutableList.<Principal>of(
+            new UserPrincipal("u3"), new GroupPrincipal("g4")));
+  }
+
+  private static Map<GroupPrincipal, Collection<Principal>> emptyGroups() {
+    return ImmutableMap.<GroupPrincipal, Collection<Principal>>of();
+  }
+
+  /**
+   * Returns the group definitions broken into batches of a batchSize entries.
+   */
+  // TODO(bmj): Uncomment @SafeVarargs when moving to Java 7 and remove the
+  // @SuppressWarnings("unchecked") usages below.
+  // @SafeVarargs
+  private static List<List<Map.Entry<GroupPrincipal, Collection<Principal>>>>
```

Comment:
I didn't mean to suggest moving this up. I would keep this and its tests buried.